src/compiler/commandLineParser.ts

```diff
@@ -652,6 +652,9 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
     {
         name: "checkJs",
         type: "boolean",
+        affectsBuildInfo: true,
+        affectsModuleResolution: true,

```

Comment:
I think only `affectsSemanticDiagnostics` is enough and doesnt need `affectsModuleResolution` any more.. even allowJs. earlier we decided if we will resolve js if allowJs is true that was long ago and its not true any more. so i dont think these flags affect module resolution?

Reply:
Ah, right, I vaguely recall us always resolving JS. Let me tweak it.

Reply:
Removed some options in follow-up commit; no baseline changes, which is good.